src/server/clientiface.cpp

```diff
@@ -343,24 +343,21 @@ void RemoteClient::GetNextBlocks (
 			const bool want_emerge = !block || !block->isGenerated();
 
 			// if the block is already in the emerge queue we don't have to check again
-			if (want_emerge && emerge->isBlockInQueue(p)) {
-				nearest_emerged_d = d;
-				continue;
-			}
-
-			/*
-				Check occlusion cache first.
-			 */
-			if (m_blocks_occ.find(p) != m_blocks_occ.end())
-				continue;
+			if (!want_emerge || !emerge->isBlockInQueue(p)) {
```

Comment:
If someone can think of a better way to express this, I am all ears.
`!(want_emerge && emerge->isBlockInQueue(p))` did not look nice either.